agent/rpcclient/health/health.go

```diff
@@ -0,0 +1,64 @@
+package health
+
+import (
+	"context"
+
+	"github.com/hashicorp/consul/agent/cache"
+	cachetype "github.com/hashicorp/consul/agent/cache-types"
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+type Client struct {
+	NetRPC NetRPC
+	Cache  CacheGetter
+}
+
+type NetRPC interface {
+	RPC(method string, args interface{}, reply interface{}) error
+}
+
+type CacheGetter interface {
+	Get(ctx context.Context, t string, r cache.Request) (interface{}, cache.ResultMeta, error)
+}
+
+func (c *Client) ServiceNodes(
+	ctx context.Context,
+	req structs.ServiceSpecificRequest,
+) (structs.IndexedCheckServiceNodes, cache.ResultMeta, error) {
```

Comment:
Another big change here that I should call out is that these no longer use pointer types. 

Generally non-pointer types should be correct for "data" objects, and request/response objects should be primarily data and not behaviour, so using a non-pointer seems correct.